dashboard/app/views/lti/v1/account_linking/landing.haml

```diff
@@ -6,21 +6,26 @@
     if landing_session.present?
       lti_provider = landing_session[:lti_provider_name]
       new_cta_type = landing_session[:new_cta_type]
+      signup_flow_type = landing_session[:signup_flow_type]
+      user_type = landing_session[:user_type]
     else
       lti_provider ||= params[:lti_provider]
       new_cta_type ||= params[:new_cta_type]
+      signup_flow_type ||= params[:signup_flow_type]
+      user_type ||= current_user&.user_type
     end
 
     email ||= params[:email]
     new_account_url = DCDO.get('student-email-post-enabled', false) ? users_finish_sign_up_path : new_user_registration_url
+    script_data[:signup_flow_type] = signup_flow_type
     script_data[:lti_provider] = lti_provider
     script_data[:lti_provider_name] = Policies::Lti::LMS_PLATFORMS[lti_provider.to_sym][:name]
     script_data[:new_account_url] = new_account_url
     script_data[:existing_account_url] = CDO.studio_url(lti_v1_account_linking_finish_link_path, CDO.default_scheme)
     script_data[:continue_account_url] = session[:user_return_to] || home_path
     script_data[:email] = email
     script_data[:new_cta_type] = new_cta_type
-    script_data[:user_type] = landing_session[:user_type] || current_user&.user_type
+    script_data[:user_type] = user_type
```

Comment:
This change fixes the case where `landing_session` is `nil` which throws an error since it can't get `user_type` in `nil`